test/integration/scheduler/filters/filters_test.go

```diff
@@ -1792,3 +1793,233 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+							PDName:    "test-disk",
+							FSType:    "ext4",
+							Partition: 0,
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node1").Label("topology.gke.io/zone", "us-west1-a").Obj(),
+			},
+			incomingPod:    st.MakePod().Name("pod1").Container(pause).PVC("pvc1").Obj(),
+			fits:           true,
+			candidateNodes: []string{"node1"},
+			csiNode: &storagev1.CSINode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node1",
+					Annotations: map[string]string{
+						v1.MigratedPluginsAnnotationKey: "kubernetes.io/gce-pd",
+					},
+				},
+			},
+		},
+		{
+			name: "pv label conflict",
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node1").Label("topology.gke.io/zone", "us-west1-a").Obj(),
+			},
+			incomingPod: st.MakePod().Name("pod1").Container(pause).PVC("pvc1").Obj(),
+			pv: v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{Name: "pv-with-csi", Labels: map[string]string{v1.LabelFailureDomainBetaZone: "us-west1-b"}},
+				Spec: v1.PersistentVolumeSpec{
+					AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteMany},
+					Capacity:    v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")},
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+							PDName:    "test-disk",
+							FSType:    "ext4",
+							Partition: 0,
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+			csiNode: &storagev1.CSINode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node1",
+					Annotations: map[string]string{
+						v1.MigratedPluginsAnnotationKey: "kubernetes.io/gce-pd",
+					},
+				},
+			},
+		},
+		{
+			name: "pv label conflicts, node has no csi plugin",
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node1").Label(v1.LabelTopologyZone, "us-west1-b").Obj(),
+			},
+			incomingPod: st.MakePod().Name("pod1").Container(pause).PVC("pvc1").Obj(),
+			pv: v1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{Name: "pv-with-csi", Labels: map[string]string{v1.LabelTopologyZone: "us-west1-a"}},
+				Spec: v1.PersistentVolumeSpec{
+					AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteMany},
+					Capacity:    v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")},
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+							PDName:    "test-disk",
+							FSType:    "ext4",
+							Partition: 0,
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+			csiNode: &storagev1.CSINode{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node1",
+					Annotations: map[string]string{
+						v1.MigratedPluginsAnnotationKey: "kubernetes.io/gce-pd",
+					},
+				},
+			},
+		},
+		{
+			name: "pv label matches, node has no csi plugin",
```

Comment:
Can you add a test case for beta label and no csi plugin?

This is an important case for cluster autoscaler, which currently does not simulate CSINode objects. I wonder if the logic we have to only translate if csi migration is turned on the node (in volumebinding + volumezone plugins) will cause this scenario to fail once we remove the beta label from nodes.

I wonder if to solve this we do just need to translate beta labels to GA labels. Unfortunately we can't assume csi migration is enabled everywhere, as we still have a few intree plugins that have not GA-ed their migration features yet.

Reply:
I've added some test cases, if the pv and the node both have the beta label with matched value, the pod will pass the filter. However if the pod has the beta label and the node has ga label with the same value, it won't pass the filter as before.

> Unfortunately we can't assume csi migration is enabled everywhere

If we want a pv with beta label matches the node with the ga label, we can translate beta labels to GA labels in `volume_zone`?

Reply:
> If we want a pv with beta label matches the node with the ga label, we can translate beta labels to GA labels in `volume_zone`?

@AxeZhan yeah I wonder if that is the simplest approach. I don't see a better way to handle the cluster autoscaler case where there is no CSINode. Apologies for all the back and forth on this. Maybe keep a copy of the current state just in case we want to come back to this approach.

Reply:
No need to apologize, I'll create another pr for this new approach. We can let this pr stay here, until we decide which approach to use.

Reply:
Well, I think we still need to let volumbebinding handle the migrable pv. So I suggest just to translate beta label to ga in this pr?(look at the new commit)
But this is breaking the old behaviour, right?

Reply:
@msau42, are you requiring any other changes or is this good to go?

Reply:
@AxeZhan sorry for the delay. Can you clarify which scenario where translating beta to GA label is insufficient?

Reply:
> @AxeZhan sorry for the delay. Can you clarify which scenario where translating beta to GA label is insufficient?

Like when a node has a label "topology.gke.io/zone" and a old pv has the beta label, if we simply translate this beta label to ga, I think volume_zone will still fail the match? Unless we translate this beta label to "topology.gke.io/zone" in volume_zone like volume_binding did? Not sure.

Reply:
@msau42

Reply:
I would expect that all nodes will have the GA labels in addition to the csi-specific labels, especially if the volume plugin was depending on the beta labels previously.

Reply:
Hmm, in that case, I think it's fair to only translate beta to GA in volume_zone, I'll create a new pr later